### ISSUE TITLE
Onyx connection can be called twice for the same data

### ISSUE THREAD

**Issue by hannojg** [Expensify/react-native-onyx#569]
There exists a "race condition" where an Onyx connection callback will be called twice for the same data. Potentially this can lead to unnecessary rerenders in the react applications.

The race condition happens when creating a connection and shortly after setting the data:

![Screenshot 2024-07-18 at 15 12 29](https://github.com/user-attachments/assets/2181d9a4-9f05-4025-b2e8-5dfe0a79a2ac)

It once gets called here in `.set`:

https://github.com/Expensify/react-native-onyx/blob/c53826bbfe31e423fad0c33b0cb9ab097b88f128/lib/Onyx.ts#L274-L275

and then here once the connection promise resolved:

https://github.com/Expensify/react-native-onyx/blob/c53826bbfe31e423fad0c33b0cb9ab097b88f128/lib/Onyx.ts#L175

**Comment by hannojg** [comment]
I think I could build a failing unit test and fix the issue (cc @mountiny)

**Comment by mountiny** [comment]
@hannojg yes please, feel free to investigate a fix for this

**Comment by hannojg** [comment]
PR is up here:

- https://github.com/Expensify/react-native-onyx/pull/570

**Comment by mountiny** [comment]
@hannojg are you able to create the onyx bump now with the checklist filled in? thanks!

**Comment by hannojg** [comment]
Hey, yes I could, but I think you mentioned me [in some other issue](https://github.com/Expensify/App/issues/37560#issuecomment-2251989745) and someone else is doing the bump, no?

**Comment by mountiny** [comment]
Yeah that was after i commented here, i think they can handle that for you

**Comment by hannojg** [comment]
My first PR [caused a few test to fail in the newdot code](https://github.com/Expensify/App/pull/46297#issuecomment-2252785350). I addressed the problems in these PRs (would appreciate a review):

- https://github.com/Expensify/react-native-onyx/pull/573
- https://github.com/Expensify/react-native-onyx/pull/574

**Comment by mountiny** [comment]
@hannojg I believe these changes are now live in the App, thank you! Do you reckon we can close this issue out?

**Comment by hannojg** [comment]
Yes, thats fixed now!
